src/hotspot/share/asm/register.hpp

```diff
@@ -245,14 +253,44 @@ inline ReverseRegSetIterator<RegImpl> AbstractRegSet<RegImpl>::rbegin() {
 
 // Debugging support
 
+template<typename R, typename... Rx>
+inline constexpr bool different_registers(AbstractRegSet<R> allocated_regs, R first_register) {
```

Comment:
different_registers is only used by debug-only code in assert_different_registers.  Shouldn't all the overloads
for different_registers be within an `#ifdef ASSERT` block?

Reply:
I could do so, but that would lose the ability to do `static_assert(different_registers(...`. I don't think that  `static_assert` depends on `ASSERT`. I'm happy to make this patch debug-only, though, if you prefer.